scripts/lib/CIME/case_st_archive.py

```diff
@@ -545,7 +555,7 @@ def _archive_process(case, archive, last_date, archive_incomplete_logs, copy_onl
                        archive_incomplete_logs, archive_file_fn)
 
     # archive restarts and all necessary associated files (e.g. rpointer files)
-    datenames = _get_datenames(case.get_value("RUNDIR"), case.get_value("CASE"))
+    datenames = _get_datenames(case.get_value("RUNDIR"), case.get_value("CASE"), case)
```

Comment:
If you are going to pass case as an argument then just pass case and get "RUNDIR" and "CASE" inside, 
or since you are only really adding multi_driver, just change the argument list to 
case.get_value("RUNDIR"), case.get_value("CASE"), case.get_value("MULIT_DRIVER")